### PR TITLE
#954 NVCS.ALWAYS and presence checker must not lead to inclusion of explicit null set

### PR DESCRIPTION
**Pull request by sjaakd**

### CODE REVIEW

**Comment by sjaakd** [comment]
I'm still in the process of adding a cleanup commit to this one.. will take some minutes.

**Comment by sjaakd** [comment]
done..  life starts to make sense again for collections  :smile:

File: processor/src/test/java/org/mapstruct/ap/test/bugs/_913/Issue913SetterMapperForCollectionsTest.java [hunk 1]
```diff
@@ -200,38 +219,110 @@ public void shouldReturnNullForCreateWithPresenceChecker() {
...
+
+        doControlAsserts( domain1, domain2 );
+        assertThat( domain1.getLongs() ).isEqualTo( domain2.getLongs() );
+        assertThat( domain1.getStrings() ).isEmpty();
+        assertThat( domain1.getLongs() ).isEmpty();
+        assertThat( domain2.getStrings() ).isEmpty();
+        assertThat( domain2.getLongs() ).isEmpty();
+    }
+
+
```
Comment (agudian) [inline processor/src/test/java/org/mapstruct/ap/test/bugs/_913/Issue913SetterMapperForCollectionsTest.java]: too many empty lines... 😁

**Comment by sjaakd** [comment]
closing this in favor of #959 (right branch)

**Pull request closed without merging**